Source/WebView/turbo.js

```diff
@@ -103,6 +103,9 @@
         if (Turbo.navigator.locationWithActionIsSamePage(location, options.action)) {
           Turbo.navigator.view.scrollToAnchorFromLocation(location)
           return
+        } else if (this.currentVisit?.location?.href === location.href) {
```

Comment:
In the corresponding Android PR, I found that the approach doesn't work immediately after a page cold boot, because `this.currentVisit` hasn't been set yet. Using `Turbo.navigator.location?.href === location.href` works in all circumstances, though. https://github.com/hotwired/turbo-android/pull/292/files#diff-aafcc0efe91961803491a50ad6158bc1b6416c69417325a6fa5d1bc50a2be45cR105